src/responder.ts

```diff
@@ -51,182 +61,357 @@ enum ResponderEvent {
+     */
+    public async sendTransaction(transactionRequest: ethers.providers.TransactionRequest): Promise<string> {
+        const txResponse = await this.signer.sendTransaction(transactionRequest);
+        return txResponse.hash;
+    }
+
+    /**
+     * Resolves after the transaction receives the first confirmation.
+     * Rejects with `NoNewBlockError` the provider does not receive a new block for `newBlockTimeout` milliseconds.
+     * Rejects with `BlockThresholdReachedError` if the transaction is still unconfirmed
+     * after `blocksThresholdForStuckTransaction` blocks are mined.
+     *
+     * @param txHash The transaction hash, returned by `sendTransaction`.
+     * @param timeLastBlockReceived Optional; if known, the time when the last block was received by the provider.
+     *                              The provider will be considered unresponsive after `newBlockTimeout` from `timeLastBlockReceived`.
+     *                              If not given, the current time returned by `Date.now()` will be used.
+     */
+    public async waitForFirstConfirmation(txHash: string, timeLastBlockReceived: number = Date.now()) {
+        const lastBlockNumberSeen = await this.signer.provider.getBlockNumber();
+
+        // Promise that waits for the first confirmation
+        const firstConfirmationPromise = waitForConfirmations(this.signer.provider, txHash, 1);
+
+        // Promise that rejects after WAIT_BLOCKS_BEFORE_RETRYING blocks are mined
+        const firstConfirmationTimeoutPromise = rejectAfterBlocks(
+            this.signer.provider, lastBlockNumberSeen, this.blocksThresholdForStuckTransaction
+        );
+
+        // ...but stop with error if no new blocks come for too long
+        const noNewBlockPromise = rejectIfAnyBlockTimesOut(
+            this.signer.provider,
+            timeLastBlockReceived,
+            this.newBlockTimeout,
+            1000
+        );
+
+        try {
+            // First, wait to get at least 1 confirmation, but throw an error if the transaction is stuck
+            // (that is, new blocks are coming, but the transaction is not included)
+            await Promise.race([
+                firstConfirmationPromise,
+                firstConfirmationTimeoutPromise,
+                noNewBlockPromise
+            ]);
+        } finally {
+            // Make sure any pending CancellablePromise is released
+            firstConfirmationPromise.cancel();
+            firstConfirmationTimeoutPromise.cancel();
+            noNewBlockPromise.cancel();
+        }
+    }
+
+
+    /**
+     * Resolves after the transaction `txHash` receives `confirmationsRequired`.
+     * Rejects with `NoNewBlockError` the provider does not receive a new block for `newBlockTimeout` milliseconds.
+     * Rejects with `ReorgError` if the transaction is not found by the provider.
+     */
+    public async waitForEnoughConfirmations(txHash: string, timeLastBlockReceived = Date.now()) {
```

Comment:
This default isnt used

Reply:
Same as above.

Reply:
Removed in 352fc270d00d7838ef20e737c027295a3bb1c97d.